app/src/main/java/net/bible/android/view/activity/readingplan/DailyReading.kt

```diff
@@ -29,282 +30,431 @@ import android.widget.Button
+    private var readingPlanMetaID: Int? = null
+    private var readingPlanDayNumber: Int? = null
+    private lateinit var readingPlanOneDay: ReadingPlanOneDayDB
+    private lateinit var layoutReadingsContainer: TableLayout
+
+    private var imageTickList: MutableList<ImageView> = ArrayList()
 
-    private lateinit var mReadings: OneDaysReadingsDto
 
-    @Inject lateinit var readingPlanControl: ReadingPlanControl
     @Inject lateinit var readingPlanActionBarManager: ReadingPlanActionBarManager
 
-    /** Called when the activity is first created.  */
     @SuppressLint("MissingSuperCall")
     override fun onCreate(savedInstanceState: Bundle?) {
-        super.onCreate(savedInstanceState, true)
+        super.onCreate(savedInstanceState, false)
         Log.i(TAG, "Displaying one day reading plan")
         setContentView(R.layout.reading_plan_one_day)
 
         super.buildActivityComponent().inject(this)
         super.setActionBarManager(readingPlanActionBarManager)
 
-        try {
-            // may not be for current day if user presses forward or backward
-            mDay = readingPlanControl.currentPlanDay
-            val extras = intent.extras
-            if (extras != null) {
-                if (extras.containsKey(PLAN)) {
-                    readingPlanControl.setReadingPlan(extras.getString(PLAN))
-                }
-                if (extras.containsKey(DAY)) {
-                    mDay = extras.getInt(DAY, mDay)
-                }
+        ABEventBus.getDefault().register(this)
+        Log.d(TAG, "Registered to ABEventBus")
+
+        layoutReadingsContainer = findViewById(R.id.reading_container)
+
+        var readingPlanInfo: ReadingPlanInformationDB? = null
+        var readingPlanOneDay: ReadingPlanOneDayDB? = null
+
+        if (dbAdapter.metaCurrentActiveReadingPlanID == null) {
+            // Load plan selection screen
+            this.startActivityForResult(
+                Intent(this, ReadingPlanSelectorList::class.java),
+                REQUEST_CODE_READING_PLAN_LIST
+            )
+
+        } else {
+
+            readingPlanInfo = ReadingPlanInformationDB(readingPlanMetaID)
+            readingPlanOneDay = ReadingPlanOneDayDB(readingPlanInfo)
+
+            loadPlanOneDay(readingPlanOneDay, readingPlanInfo)
+
+        }
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        ABEventBus.getDefault().unregister(this)
+        Log.d(TAG, "Unregistered from ABEventBus")
+    }
+
+    /**
+     * Function loads reading plan information screen with correct data
+     * @param readingPlanOneDayParam Allows nulls, but the ONLY time
+     * null should be passed is when no reading plan has been started.
+     * @param readingPlanInformationParam Is REQUIRED ONLY when the
+     * [readingPlanOneDayParam] is passed as NULL
+     */
+    private fun loadPlanOneDay(
+        readingPlanOneDayParam: ReadingPlanOneDayDB?,
+        readingPlanInformationParam: ReadingPlanInformationDB? = null,
```

Comment:
The reason I put the Param suffix at a few places is because I needed to use another variable throughout the function. In this case line 150, might be 154 for you:
`readingPlanOneDay.readingPlanInfo = readingPlanInformationParam`

In this case I'm setting the readingPlanInfo as readingPlanInformationParam. I'm trying to keep them distinguished and not have the same names.

Reply:
Okay. Let's keep this like this for a while. Maybe I'll comment this later again. Though somehow I don't like it but don't know what to suggest either. 

Some options.
- You could have same name, and if you need to refer to the attribute within same context, you would use `this.readingPlanInformation`
- You could use `_` suffix (i.e. `readingPlanInfo_`). I think I have used that sometimes. (though I must admit that this convention comes from Python world)

Reply:
Done